agent/job_runner.go

```diff
@@ -75,6 +75,21 @@ type JobRunnerConfig struct {
 	// The configuration of the agent from the CLI
 	AgentConfiguration AgentConfiguration
 
+	// The logger to use
+	Logger logger.Logger
```

Comment:
If there's already logger in `JobRunner`, which contains `JobRunnerConfig`, why is there also one in `JobRunnerConfig`. I think we need to chose definitions of the structs where this sort of thing does not happen.